students/Michael Hrstka/Lab 1/RGB_LED/RGB_LED.ino

```diff
@@ -0,0 +1,53 @@
+int ledR = 3;
+int ledG = 5;
+int ledB = 6;
+int btn = 12;
+int state = 0;
+
+void setup() {
+  pinMode(ledR, OUTPUT);
+  pinMode(ledG, OUTPUT);
+  pinMode(ledB, OUTPUT);
+  pinMode(btn, INPUT);
+}
+
+void chkBtn() {
+  if (digitalRead(btn) == HIGH) {
+    state = state > 4 ? 0 : state + 1;
+        while (digitalRead(btn) == HIGH){ // Debounce!
+      delay(10);
```

Comment:
Interesting strategy. But I imagine if you hold the button you'll pause whatever the LED is doing. E.g., the fading probably stops while you hold the button down. Is that desired/expected behavior?

Reply:
Expected but not necessarily desired.